### ISSUE TITLE
[Avatar] Sizes variant has badge what does not have `role` specified

### ISSUE THREAD

**Issue by a11ymiko** [sl-design-system/components#3013]
## Current behavior
In [Avatar Sizes](https://jolly-stone-0f7fffc03-3000.westeurope.6.azurestaticapps.net/?path=/story/media-avatar--sizes) every `<sl-badge>` do not have `role` specified, which makes it forbidden to use `aria-label` on `<sl-badge>`.
<img width="309" height="566" alt="Image" src="https://github.com/user-attachments/assets/4939f7c3-24a8-4a09-8d72-53c5ffffc88c" />

## Proposed solution
Adding `role="text"` to each `<sl-badge>` should fix the issue that is reported by a11y add-on in Storybook. 

### 👤 Your name

Kacper

### 🧢 Your Product/Team

SLDS
